### ISSUE TITLE
ticketing system

### ISSUE THREAD

**Issue by helenvarley** [upstage-org/mobilise#93]
As an artist creating a performance, I expect to be able to sell tickets via an online ticketing system, and have a way to manage this so that audience members can enter the stage with a prepaid ticket, buy a ticket "at the door", or have a complimentary ticket.

probably integration with an existing 3rd party platform is the way to do this? can artists choose their preferred platform from a number of options? eg:

- https://swicket.io/
- https://github.com/Attendize/Attendize
- https://www.goodfirms.co/blog/best-free-open-source-ticketing-software

**Comment by paul-rohrlach** [comment]
This is a great idea.  Suggest we put this as a low-priority for now though.  It's not needed for the upcoming performances, right?

**Comment by helenvarley** [comment]
no; it came up because vicki & i attended the Ideathon yesterday & we are putting in a proposal (the cash prizes are not huge, but there is mentoring & networking etc that could be beneficial) & as part of that we have to do a business plan; we also have to do a business plan as part of the EU project. so we were brainstorming possible revenue streams & ticketing is obviously one of them.

... ideally it should be something that people can choose to use or not, & something that gives a percentage to UpStage & a percentage to the artists. typically in physical theatres it would be a 70/30 - 80/20 split, the greater share going to the artists & the smaller share going to the venue, who would also provide the ticketing system & collect the money. however if artists are chosing their own 3rd party ticketing platform i'm not sure how we insist on getting a percentage from them. might need to be that UpStage signs up with one or more ticketing systems & collects the cash then pays out to the artists (as the theatre venues do). plenty to think about - but definitely doesn't have to be thought about just yet :)

**Comment by paul-rohrlach** [comment]
This will definitely require extra planning, etc.

Financial systems are... Tricky 😉 .

**Comment by helenvarley** [comment]
yes; but hopefully we can use 3rd party platforms as much as possible & just provide a connecting doorway?

**Comment by vickis** [comment]
yes indeed Paul
it is a discussion that has come up over the years but now more than ever
people are accepting that just because its online doesn't mean it costs
nothing to produce
in our discussion for hopefully the TechCul assistance for the business
training model we are framing the support for at how this might work well
and still enable our creative and less financial community to take part
the links were from a quick google search
would be super to have systems in place for how people might support
UpStage :)

**Comment by helenvarley** [comment]
i am setting up a ticketing system for another project and we were going to use eventyay (https://eventyay.com/) which is open source & a project of FOSSAsia; they've been using it for events that we've been part of in the last 6 months & it seemed good. however, after initially getting a fast response when asking for support, it's now been nearly a week with no response to some quite major questions, and i've discovered that the support material on the website is pretty lacking. we have a test event and no-one has received an email of their ticket, they have to go into their eventyay dashboard to find the ticket, and it's not possible to customise the ticket at all so it's actually not very nice looking.

i'm now re-looking at pretix, https://pretix.eu/, which is also open source and has the added advantage of being available in german ;) it seems to be a bit more nicely designed and less geeky. this is the system used by the stuttgart filmwinter where we have presented UpStage this year & last year; i asked them about it & they had no complaints.

this is still not a high priority but i'm just noting these things down here while they are on my mind. i think @gloriajw was looking at eventyay, don't know what opinion of it she formed ...

**Comment by vickis** [comment]
Pretix sounds better :)

**Comment by helenvarley** [comment]
coming back to this issue again :)

what we need is a way that artists can use their own third party ticketing system to sell tickets to UpStage performances. 
we (as in UpStage Live) do not want to be handling ticketing for everyone's shows; but there needs to be some integration so that artists can control who enters ticketed shows.

- can it be possible for artists to use any ticketing system of their choice, or will we have to select one or a small number that UpStage integrates with?
- can it work so that our clients (e.g. meta, refinery) can manage it all themselves, we don't have to do anything extra for their ticketing? this would obviously be ideal.

**Comment by gloriajw** [comment]
- can it be possible for artists to use any ticketing system of their
   choice, or will we have to select one or a small number that UpStage
   integrates with? (No, and yes)
   - can it work so that our clients (e.g. meta, refinery) can manage it
   all themselves, we don't have to do anything extra for their ticketing?
   this would obviously be ideal. (If we roll our own and don't accept money,
   maybe.

**Comment by helenvarley** [comment]
i guess the fundamental things are to minimise the work on our side, and definitely not to handle transactions for others' ticketing. but i have already had people asking me how artists can monetise their work through UpStage, so we need a solution.

**Comment by helenvarley** [comment]
i just noticed in a FOSSASIA email that their ticketing system Eventyay is about to get a major upgrade - maybe it's worth looking at again?
https://github.com/orgs/fossasia/repositories?language=&q=eventyay
https://www.eventyay.com/

**Comment by helenvarley** [comment]
this is an old issue, but it still needs to be eventually achieved 😆  
it may well be that it can be achieved through upstage.org.nz as there are now quite a lot of wordpress plugins that can handle ticketing. 
this would then need to be integrated so that only those who have purchased a ticket can enter the stage - some kind of code to be entered at the stage loading point or similar. 
vicki & i are going to look into ticketing as part of the business plan for UpStage that we are hopefully doing over the rest of this year.
